### PR TITLE
MCR-2200 add filter to enable content negotation by extension

### PR DESCRIPTION
**Pull request by rsteph-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2200).

### CODE REVIEW

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5511336&bid=18062296)

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on 7beeec0710e844bfde1b376460e515a54b1bfbea! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5520-pull_request-964

File: mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * 
+ */
+
+@Provider
+@PreMatching
+public class MCRContentNegotiationViaExtensionFilter implements ContainerRequestFilter {
+
+    private static final Map<String, MediaType> MEDIA_TYPE_MAPPINGS = Map.ofEntries(
+            new AbstractMap.SimpleEntry<String, MediaType>("json", MediaType.APPLICATION_JSON_TYPE),
+            new AbstractMap.SimpleEntry<String, MediaType>("xml", MediaType.APPLICATION_XML_TYPE));
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java]: should be `Map.entry(key, value)`

File: mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    private UriConnegFilter uriConnegFilter = new UriConnegFilter(MEDIA_TYPE_MAPPINGS, Map.of());
+
+    @Override
+    public void filter(ContainerRequestContext requestContext) throws IOException {
+        if(!requestContext.getUriInfo().getPath().contains("/contents/")) {
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java]: Better use a property with regex filter

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestV2App.java [hunk 3]
```diff
@@ -50,8 +51,11 @@
 
     public MCRRestV2App() {
         super();
+        if (MCRConfiguration2.getBoolean("MCR.RestAPI.V2.EnableContentNegotiationViaFileExtension").orElse(false)) {
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestV2App.java]: We should always enable it (can be excluded by regex: see above)

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java (hunk 1 above)
Comment (rsteph-de) [reply mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java]: O.K.

File: mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java (hunk 2 above)
Comment (rsteph-de) [reply mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java]: done

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestV2App.java (hunk 3 above)
Comment (rsteph-de) [reply mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestV2App.java]: done

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5511336&bid=18062296)

**Comment by yagee-de** [comment]
How does this Filter interfere with MCRNormalizeMCRObjectIDsFilter?

**Comment by rsteph-de** [comment]
> 
> 
> How does this Filter interfere with MCRNormalizeMCRObjectIDsFilter?

Conceptually it should not. The extension *.xml / *.json should be removed first and after that the  MCRNormalizeMCRObjectIDsFilter should be applied. - I hope the code "works as designed".
Let's check the order of filters again.

**Comment by adlerre** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- mycore-restapi/src/main/java/org/mycore/restapi/MCRContentNegotiationViaExtensionFilter.java  2
- mycore-restapi/src/main/java/org/mycore/restapi/MCRNormalizeMCRObjectIDsFilter.java  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mycore/pullRequest?prid=5511336&bid=18062296)

**Review by yagee-de — APPROVED** [review]

**Pull request merged**